enterprise/cmd/frontend/internal/auth/azureoauth/provider.go

```diff
@@ -84,10 +85,14 @@ func parseConfig(logger log.Logger, cfg conftypes.SiteConfigQuerier, db database
 			continue
 		}
 
-		// Currently Azure Dev Ops will work only against https://dev.azure.com. If we have more
-		// than one configuration for Azure Dev Ops auth provider, we want to fail early.
+		var configured bool
+		for _, clientID := range configuredIDs {
+			if clientID == pr.AzureDevOps.ClientID {
+				configured = true
+			}
+		}
```

Comment:
As much as I dislike this code and wish there was a builtin `Contains` or similar on the slice, we should probably break out of the loop when we find a match.

```suggestion
		for _, clientID := range configuredIDs {
			if clientID == pr.AzureDevOps.ClientID {
				configured = true
				break
			}
		}
```

Unless we use a `Set` as suggested above.